src/main/java/com/cloudbees/jenkins/support/config/SupportAutomatedBundleConfiguration.java

```diff
@@ -0,0 +1,252 @@
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import static java.util.logging.Level.WARNING;
+
+/**
+ * Global Configuration for the Automated Support Bundle Generation, see {@link SupportPlugin.PeriodicWorkImpl}.
+ * 
+ * @author Allan Burdajewicz
+ */
+@Extension
+@Restricted(NoExternalUse.class)
+@Symbol("automatedBundleConfiguration")
+public class SupportAutomatedBundleConfiguration extends GlobalConfiguration {
+
+    private static final Logger LOG = Logger.getLogger(SupportAutomatedBundleConfiguration.class.getName());
+
+    /**
+     * Gets the singleton instance.
+     *
+     * @return Singleton instance
+     */
+    public static @NonNull
+    SupportAutomatedBundleConfiguration get() {
+        return ExtensionList.lookupSingleton(SupportAutomatedBundleConfiguration.class);
+    }
+
+    /**
+     * The list of components configured for background generation.
+     */
+    private List<String> componentIds;
+
+    /**
+     * Whether to enable Support Bundle background generation or not.
+     */
+    private boolean enabled = true;
+
+    /**
+     * How often automatic support bundles should be collected. Should be {@code 1} unless you have very good reason
+     * to use a different period. {@code 0} disables bundle generation and {@code 24} is the longest period permitted.
+     */
+    private int period = 1;
+
+    public SupportAutomatedBundleConfiguration() {
+        load();
+    }
+
+    public List<String> getComponentIds() {
+        return componentIds == null
+            ? getDefaultComponentIds()
+            : componentIds;
+    }
+
+    /**
+     * Get the default list of Component Ids for the automated bundle generation.
+     *
+     * @return a list of {@link String}
+     */
+    public static List<String> getDefaultComponentIds() {
+        return getApplicableComponents()
+            .stream()
+            .filter(Component::isEnabled)
+            .filter(Component::isSelectedByDefault)
+            .map(Component::getId)
+            .collect(Collectors.toList());
+    }
+
+    public boolean isEnabled() {
+        return enabled;
+    }
+
+    public int getPeriod() {
+        return period;
+    }
+
+    /**
+     * Return if the {@link SupportAutomatedBundleConfiguration#period} is enforced by the System Property
+     * {@link SupportPlugin#AUTO_BUNDLE_PERIOD_HOURS}.
+     *
+     * @return a list of {@link String}
+     */
+    public boolean isEnforcedPeriod() {
+        return System.getProperty(SupportPlugin.class.getName() + ".AUTO_BUNDLE_PERIOD_HOURS") != null;
+    }
+
+    @DataBoundSetter
+    public void setComponentIds(List<String> componentIds) {
+        this.componentIds = componentIds;
+    }
+
+    @DataBoundSetter
+    public void setEnabled(boolean enabled) {
+        if (isEnforcedDisabled()) {
+            this.enabled = SupportPlugin.AUTO_BUNDLE_PERIOD_HOURS >= 1;
```

Comment:
This is never going to happen because the if is checking that the period hours is <= 0.

Reply:
The value 0 can happen. This is the current way that users have to disable the automatic generation. After double checking, values less than 0 cannot happen because of the `min` and `max` used when we initialize from the system property: https://github.com/jenkinsci/support-core-plugin/blob/support-core-2.70/src/main/java/com/cloudbees/jenkins/support/SupportPlugin.java#L132-L137.

Reply:
What I mean is: if "isEnforceDisabled()" return true, then the value is 0 and so this.enabled is always going to be false.

Reply:
Oh I see. You're right!